src/main/java/org/apache/commons/collections4/IterableUtils.java

```diff
@@ -50,6 +50,11 @@
  */
 public class IterableUtils {
 
+    /**
+     * {@code IterableUtils} should not normally be instantiated.
+     */
+    private IterableUtils() {}
+
```

Comment:
This is a public class that used to have a public (default) constructor. You have made it private which breaks binary compatibility.

The other class `ArrayUtils` where you added a private constructor is a package private class and so that change does not break binary compatibility which compares the API at public and protected scope.

Reply:
Hi @aherbert 

Yep. I know. and that's my intentions. Unify the constructor.

Reply:
There is nothing planned for 5.0 AFAIK, 4.5 would be next. Creating a 5.0 branch now would be premature IMO and I personally would not want to spend the time maintaining it.

Reply:
Perhaps make the constructor public and mark it as deprecated for removal in 5.0 with a note that the class is not intended to be instantiated. This will clarify the intention and ensure the change does happen when 5.0 is released as all deprecated tags will be addressed on a major version increment.

Or remove this from the PR as binary compatibility is an important goal of commons within major versions, even for something trivial like this where the use cases of instances of this class are non-existent and I think it unlikely that any end user would be creating instances.

Reply:
Hi,

I have reverted the changes  of the class IterableUtils. Although I think it is necessary to match all the constructors to private access

Reply:
>Perhaps make the constructor public and mark it as deprecated for removal in 5.0 with a note that the class is not intended to be instantiated. This will clarify the intention and ensure the change does happen when 5.0 is released as all deprecated tags will be addressed on a major version increment.

+1 and will remember us that this can be done by 5.0.

Reply:
Thank you @kinow